earthaccess/search.py

```diff
@@ -313,35 +342,23 @@ def get(self, limit: int = 2000) -> list:
 
     def temporal(
         self,
-        date_from: Optional[Union[str, dt.datetime]] = None,
-        date_to: Optional[Union[str, dt.datetime]] = None,
+        date_from: Optional[Union[str, dt.date]] = None,
+        date_to: Optional[Union[str, dt.date]] = None,
```

Comment:
From a typing standpoint, using `dt.date` here is correct as `dt.datetime` is a subclass of `dt.date`, but I wonder, from a documentation perspective, if it makes sense to also list datetime? That is, I'm not sure how widely recognized it is that `dt.date` covers both `dt.date` and `dt.datetime`... what do you think?
```suggestion
        date_from: Optional[Union[str, dt.date, dt.datetime]] = None,
        date_to: Optional[Union[str, dt.date, dt.datetime]] = None,
```

Reply:
Introducing logical redundancy into code (albeit, not the runtime code) for the purpose of improving documentation does two things poorly. The redundancy (a datetime is a date) but also the documentation gets the literal type hint of `Optional[Union[str, date, datetime]]` that requires  knowing about typing. I think in the future I would override the automatic document generation with the docstring ... at least, I believe that is possible.